### PR TITLE
Added tests for partition dns/pq

### PR DESCRIPTION
**Pull request by wilkermichael**
Changes proposed in this PR:
- There was a bug we were seeing previously which caused DNS queries to not work when specifying a sameness group and a partition. After manual testing with both agentless and agentful scenarios, this appears to not be happening anymore. I added some tests to exercise the `sg` + `ap` dns scenario, they are all passing.
- Did some refactoring
   - added receivers where we were previously passing in the cluster under test
   - added the concept of a `primary cluster` in order to enable dns testing for partitions

The diff shows more changes than were actually changed because of moving things around. I added some comments to the code to help you navigate a bit easier.

How I've tested this PR:
- added tests

How I expect reviewers to test this PR:
- :eyes

Checklist:
- [ ] Tests added
- [ ] [CHANGELOG entry added](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: acceptance/tests/sameness/sameness_test.go
Comment (wilkermichael) [inline acceptance/tests/sameness/sameness_test.go:624]: A lot of these functions are unchanged, I mostly refactored to use receivers instead of passing in multiple values + a cluster

File: acceptance/tests/sameness/sameness_test.go
Comment (wilkermichael) [inline acceptance/tests/sameness/sameness_test.go:153]: This is where the "meat" of the refactor changes are. The partition cluster (`cluster-01b`) needs to use the primary cluster for DNS operations.

File: acceptance/tests/sameness/sameness_test.go
Comment (wilkermichael) [inline acceptance/tests/sameness/sameness_test.go:564]: Removed the need for this check

File: acceptance/tests/sameness/sameness_test.go
Comment (wilkermichael) [inline acceptance/tests/sameness/sameness_test.go:667]: example of where the primary cluster is used

**Comment by wilkermichael** [comment]
@zalimeni I just added some comments to help you navigate. This should be ready for review now whenever you have time.

**Review by zalimeni — APPROVED** [review]
LGTM! After reordering fns locally to simplify the diff, it seems like the only functional change to helpers was the use of `primaryCluster` for DNS.

Meta-feedback, I've found it can help to split reordering/renames into separate commits from other changes so the diff is easier to scan visually commit-by-commit. Even though you called it out in the description (thank you), that could help speed review a bit more.

Thank you for the extra comments and clear PR description as guidance!

**Review by hashi-derek — APPROVED** [review]

File: acceptance/ci-inputs/kind_acceptance_test_packages.yaml
Comment (hashi-derek) [inline acceptance/ci-inputs/kind_acceptance_test_packages.yaml:2]: Just a reminder to remove these things before merging

**Pull request merged**
